### PR TITLE
bmp

### PR DESCRIPTION
**Pull request by JulKaltenegger**
This is a new namespace for the Building Material Performance Ontology.

### CODE REVIEW

**Comment by dgarijo** [comment]
There is no redirection. Please add one, we don't do namespace reservation

File: bmp/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# BMP_BuildingMaterialPerformance
+
+The Building Material Performance ontology classifies construction materials based on categories, types and property groups. Each property group contains multiple material parameters that are based on building physics and thus enable building information models to be enriched with highly aggregated information with respect to material information modelling and material simulation.
```
Comment (TallTed) [inline bmp/README.md:3]:
```suggestion
The Building Material Performance ontology classifies construction materials based on categories, types, and property groups. Each property group contains multiple material parameters that are based on building physics and thus enable building information models to be enriched with highly aggregated information with respect to material information modelling and material simulation.
```

File: bmp/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+
+Web Documentation
+https://julkaltenegger.github.io/bmp/
```
Comment (TallTed) [inline bmp/README.md:6]:
```suggestion
## Web Documentation

https://julkaltenegger.github.io/bmp/ 

## Maintainer

@julkaltenegger
```

**Review by TallTed — CHANGES_REQUESTED** [review]

**Comment by JulKaltenegger** [comment]
Hello there,

I am trying to figure out what to put into the .htaccess file.  I want to
publish the entire documentation of the ontology to a later stage, when
publishing the paper.
Do you have a reference htaccess doc?

Thanks
Julia

Am Di., 25. Juni 2024 um 17:56 Uhr schrieb Ted Thibodeau Jr <
***@***.***>:

**Comment by dgarijo** [comment]
Here is an example: https://github.com/perma-id/w3id.org/tree/master/example

**Comment by JulKaltenegger** [comment]
Hey,

My apologies for the delay..
I try to understand how this works, Even though I see the reference, I dont
understand if I have to make changes inside the .htaccess or do I simply
c/p in my repo?
I send you also a link of my own repo with the bmp ontology
https://github.com/JulKaltenegger/bmp/tree/main
so you can see that i did make this index.html already, is this the same
thing?

Thank you for clarifying.
Best regards
Julia


Am Mi., 26. Juni 2024 um 11:20 Uhr schrieb Daniel Garijo <
***@***.***>:

**Comment by davidlehn** [comment]
w3id.org is only a redirection service.  You need a `.htaccess` here to map requests to redirection responses to your site(s).  To get a minimum mapping so that a browser would send links like `https://w3id.org/bmp#Hempcrete` to that anchor on your doc site, use something like this:
```
RewriteEngine on
RewriteRule ^$ https://julkaltenegger.github.io/bmp/ [R=302,L]
```
If you look around at that above example or many of the other directories you can see examples of redirects based on request type and sub-paths and lots of other things.  Your use case determines what you will want to setup.

**Comment by JulKaltenegger** [comment]
Finally managed to add .htaccess
Can you check please and tell me if it is good?

Thanks
Julia




Am Mi., 14. Aug. 2024 um 22:44 Uhr schrieb David I. Lehn <
***@***.***>:

File: bmp/.htaccess [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+RewriteCond %{HTTP_ACCEPT} text/\* [OR]
+RewriteCond %{HTTP_ACCEPT} \*/turtle 
+RewriteRule ^$https://julkaltenegger.github.io/bmp/406.html [R=303,L]
+
+RewriteCond %{HTTP_ACCEPT} .+
+RewriteRule ^$ https://julkaltenegger.github.io/bmp/406.html[R=406,L]
+# Default response
+# ---------------------------
+# Rewrite rule to serve the RDF/XML content from the vocabulary URI by default
+RewriteRule ^$ https://julkaltenegger.github.io/bmp/406.xml [R=303,L]
```
Comment (dgarijo) [inline bmp/.htaccess:47]: Careful, this does not exist. I think you want to add https://julkaltenegger.github.io/bmp/ontology.xml instead

File: bmp/.htaccess [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+# Rewrite rule to serve N-Triples content from the vocabulary URI if requested
+RewriteCond %{HTTP_ACCEPT} application/n-triples
+RewriteRule ^$ https://julkaltenegger.github.io/bmp/ontology.nt [R=303,L]
+
+# Rewrite rule to serve TTL content from the vocabulary URI if requested
+RewriteCond %{HTTP_ACCEPT} text/turtle [OR]
+RewriteCond %{HTTP_ACCEPT} text/\* [OR]
+RewriteCond %{HTTP_ACCEPT} \*/turtle 
+RewriteRule ^$https://julkaltenegger.github.io/bmp/406.html [R=303,L]
```
Comment (dgarijo) [inline bmp/.htaccess:40]: This is not the right redirection. You may want to redirect to `https://julkaltenegger.github.io/bmp/ontology.ttl ` instead

Comment (dgarijo) [reply bmp/.htaccess:40]: Also leave a space between `^$` and `https`

**Comment by dgarijo** [comment]
@JulKaltenegger I merged but there are a few typos which I highlighted in my review. Please have a look

**Comment by JulKaltenegger** [comment]
Hey,

Thanks for the check and response. I updated the file and committed again.
Could you check again if it is good to go?






Am Di., 3. Sept. 2024 um 15:32 Uhr schrieb Daniel Garijo <
***@***.***>:

**Comment by dgarijo** [comment]
you need to open a new PR

**Comment by JulKaltenegger** [comment]
What do you mean by PR?\





Am Mi., 4. Sept. 2024 um 12:29 Uhr schrieb Daniel Garijo <
***@***.***>:

**Comment by dgarijo** [comment]
you opened a pull request (PR), and we merged it. Now if you added more changes, you  need to open a new one.

**Pull request merged**